packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -19,6 +19,11 @@ import 'theme.dart';
 /// This sample shows by dragging the user can reorder the items of the list.
 /// The [onReorder] parameter is required and will be called when a child
 /// widget is dragged to a new position.
+/// If oldIndex is before newIndex,
```

Comment:
We might as well be upfront about this and point out that typically the `onReorder` callback adjusts its index parameters like this:
```dart
      onReorder: (int oldIndex, int newIndex) {
          if (oldIndex < newIndex) {
            newIndex -= 1;
          }
          ....
        }
```
The example is also helpful however it's confusing because the list items' background colors change when they're reordered.  Also because the list items have an index property which is unrelated to the old/new index properties being assed to the onReorder callback. It would be easier to understand the example if each list item had a color and a "value" that was obviously not the item's current list index. Could just be the initial position of the item * 10. You can generate a range of colors with Color.lerp, for example `Color.lerp(Colors.red, Colors.blue, n / N)`, where 0 < n < N.

Reply:
Thanks! I was testing out this example but it actually looks kinda wonky since the background color doesn't change on the tiles (besides the closest one) while a tile is being dragged. What do you think?

https://dartpad.dev/490a83543037c0decb430b0af0ab3f5c

Reply:
I agree, that still looks confusing. Will take a closer look as soon as I get a chance.

Reply:
Sounds good!